gematria/granite/python/token_graph_builder_model.py

```diff
@@ -340,3 +358,76 @@ def _create_graph_network_modules(
             residual_connection=options.EnableFeature.BY_FLAG,
         ),
     )
+
+  def _make_batch_feed_dict(self) -> model_base.FeedDict:
+    feed_dict = super()._make_batch_feed_dict()
+    to_feed_instruction_annotations = None
+    if len(self._batch_graph_builder.instruction_annotations) == 0:
+      to_feed_instruction_annotations = np.zeros((0, 0), dtype=self.dtype)
+    else:
+      to_feed_instruction_annotations = np.array(
+          list(self._batch_graph_builder.instruction_annotations.values()),
```

Comment:
This may be dangerous: Python dicts do have a stable iteration order (iteration is in the insertion order), but this doesn't hold for C++ unordered containers, and we might be getting the values in a random order each time we run the code.

I'd prefer manually sorting columns of the tensor by a provided list of annotations (perhaps even in the C++ code, for more efficiency).